vaadin-text-field-flow-parent/vaadin-text-field-flow/src/main/java/com/vaadin/flow/component/textfield/IntegerField.java

```diff
@@ -136,41 +177,73 @@ public IntegerField(String label, Integer initialValue,
     }
 
     /**
-     * Sets the minimum value of the field. Entering a value which is smaller
-     * than {@code min} invalidates the field.
+     * {@inheritDoc}
+     * <p>
+     * Distinct error messages for unparsable input and different constraints
+     * can be configured with the {@link IntegerFieldI18n} object, using the
+     * respective properties. However, note that the error message set with
+     * {@link #setErrorMessage(String)} will take priority and override any i18n
+     * error messages if both are set.
+     */
+    @Override
+    public void setErrorMessage(String errorMessage) {
+        super.setErrorMessage(errorMessage);
+    }
+
+    /**
+     * @see IntegerFieldI18n#setRequiredErrorMessage(String)
+     */
+    @Override
+    public void setRequiredIndicatorVisible(boolean required) {
+        super.setRequiredIndicatorVisible(required);
+    }
+
+    /**
+     * Sets the minimum value for the field. This will configure the field to
+     * invalidate if the entered value is below the minimum. It will also limit
+     * the decrementing of the value when step buttons are enabled.
+     * <p>
+     * The minimum value is inclusive.
      *
      * @param min
-     *            the min value to set
+     *            the minimum integer value to set
+     * @see IntegerFieldI18n#setMinErrorMessage(String)
      */
     public void setMin(int min) {
         super.setMin(min);
     }
 
     /**
-     * Gets the minimum allowed value of the field.
+     * Gets the minimum value of the field. The constraint activates only if the
```

Comment:
Consider using "for the field" consistently:

```suggestion
     * Gets the minimum value for the field. The constraint activates only if the
```

Reply:
Done.